pkg/utils/history_buffer.go

```diff
@@ -0,0 +1,46 @@
+package utils
+
+import "fmt"
+
+type HistoryBuffer[T any] struct {
```

Comment:
Could you add a unit test or two for this struct? e.g. push a couple items, check the state of the items slice

Reply:
sure thing.

added here: eeab1e6dd94614db83addc6f98163e02b834551c